src/pgpmailer.js

```diff
@@ -46,100 +44,83 @@
      * @param {Array} options.mail.attachments (optional) Array of attachment objects with filename {String}, content {Uint8Array}, and mimeType {String}
      * @param {Boolean} options.mail.encrypted Indicating if the mail is already encrypted
      * @param {Array} options.publicKeysArmored The public keys with which the message should be encrypted
-     * @param {Function} callback(error) Indicates that the mail has been sent, or gives information in case an error occurred.
+     *
+     * * @return {Promise<String>} Resolves with the mail source when the mail has been sent
      */
-    PgpMailer.prototype.send = function(options, callback) {
+    PgpMailer.prototype.send = function(options) {
         var self = this;
 
-        if (options.encrypt) {
-            if (!options.mail.encrypted) {
-                self._pgpbuilder.encrypt(options, function(error) {
-                    if (error) {
-                        callback(error);
-                        return;
-                    }
+        return new Promise(function(resolve, reject) {
+            if (options.encrypt) {
+                if (!options.mail.encrypted) {
+                    self._pgpbuilder.encrypt(options).then(function() {
+                        resolve(self._pgpbuilder.buildEncrypted(options));
+                    }).catch(reject);
+                    return;
+                }
 
-                    self._pgpbuilder.buildEncrypted(options, onBuildFinished);
-                });
+                resolve(self._pgpbuilder.buildEncrypted(options));
                 return;
             }
 
-            self._pgpbuilder.buildEncrypted(options, onBuildFinished);
-            return;
-        }
+            resolve(self._pgpbuilder.buildSigned(options));
```

Comment:
No Need to wrap these call into a new Promise since all cases return a promise anyway.